src/Stratis.Bitcoin/P2P/PeerConnector.cs

```diff
@@ -191,27 +184,22 @@ public void StartConnectAsync()
 
             this.asyncLoop = this.asyncLoopFactory.Run($"{this.GetType().Name}.{nameof(this.ConnectAsync)}", async token =>
             {
-                await this.ConnectAsync();
+                if (!this.peerAddressManager.Peers.Any())
+                    return;
+
+                if (this.ConnectedPeers.Count >= this.MaximumNodeConnections)
+                    return;
+
+                await this.OnConnectAsync();
```

Comment:
according to gentleman agreement made earlier somewhere, we do use .ConfigureAwait(false) with all awaits in all FN code (not tests though)

Reply:
Done 👍